src/Bridge/Laravel/IlluminateValidator.php

```diff
@@ -0,0 +1,181 @@
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+// phpcs:disable
+/**
+ * Overridden so that we can cache rule parsing.
+ *
+ * @SuppressWarnings(PHPMD) This file is almost a direct copy from Lumen.
+ *
+ * // phpcs:ignore
+ */
+class IlluminateValidator extends BaseValidator
+{
+    /**
+     * Holds a static rule cache used across all validator instances.
+     *
+     * @var mixed[]
+     */
+    protected static $ruleCache = [];
+
+    /**
+     * Validate a given attribute against a rule.
+     *
+     * This method is fully overridden so we can intercept the rule parsing process
+     * and resolve it from a cache.
+     *
+     * @codeCoverageIgnore The code in this method is a direct copy and paste from Laravel.
+     *
+     * @noinspection PhpMissingParentCallCommonInspection
+     *
+     * @param string $attribute
+     * @param string $rule
+     *
+     * @return void
+     */
+    protected function validateAttribute($attribute, $rule): void
+    {
+        $this->currentRule = $rule;
+
+        [$rule, $parameters] = $this->getParsedRule($rule);
+
+        if ($rule === '') {
+            return;
+        }
+
+        // First we will get the correct keys for the given attribute in case the field is nested in
+        // an array. Then we determine if the given rule accepts other field names as parameters.
+        // If so, we will replace any asterisks found in the parameters with the correct keys.
+        $keys = $this->getExplicitKeys($attribute);
+        if (\count($keys) > 0 && $this->dependsOnOtherFields($rule)) {
+            $parameters = $this->replaceAsterisksInParameters($parameters, $keys);
+        }
+
+        $value = $this->getValue($attribute);
+
+        // If the attribute is a file, we will verify that the file upload was actually successful
+        // and if it wasn't we will add a failure for the attribute. Files may not successfully
+        // upload if they are too large based on PHP's settings so we will bail in this case.
+        if ($value instanceof UploadedFile === true && $value->isValid() === true &&
+            $this->hasRule($attribute, \array_merge($this->fileRules, $this->implicitRules)) === true
+        ) {
+            $this->addFailure($attribute, 'uploaded', []);
+
+            return;
+        }
+
+        // If we have made it this far we will make sure the attribute is validatable and if it is
+        // we will call the validation method with the attribute. If a method returns false the
+        // attribute is invalid and we will add a failure message for this failing attribute.
+        $validatable = $this->isValidatable($rule, $attribute, $value);
+
+        if ($rule instanceof RuleContract) {
+            $this->validateUsingCustomRule($attribute, $value, $rule);
+
+            return;
+        }
+
+        $method = "validate{$rule}";
+
+        if ($validatable === true && $this->$method($attribute, $value, $parameters, $this) !== true) {
```

Comment:
Should we do a `\method_exists()` check to avoid a PHP Error calling an undefined method?